src/genie/libs/parser/iosxe/show_wlan.py

```diff
@@ -1344,3 +1343,310 @@ def cli(self, output=None):
+# ==============================
+class ShowWlanIdClientStats(ShowWlanIdClientStatsSchema):
+    """Parser for show wlan id client stats"""
+
+    cli_command = "show wlan id {id_number} client stats"
+
+    def cli(self, id_number, output=None):
+        if output is None:
+            cmd = self.cli_command.format(id_number=id_number)
+            out = self.device.execute(cmd)
+
+        else:
+            out = output
+
+        # Wlan Profile Name: lizzard_Global, Wlan Id: 17
+        # Current client state statistics:
+        # -----------------------------------------------------------------------------
+        # Authenticating         : 7
+        # Mobility               : 0
+        # IP Learn               : 0
+        # Webauth Pending        : 0
+        # Run                    : 2
+
+        # Total client delete reasons
+        # ---------------------------
+        #     No Operation                                                    : 0
+        #     Internal error                                                  : 0
+        #     Deauthentication or disassociation request                      : 0
+        #     Session Manager                                                 : 0
+        #     L3 authentication failure                                       : 0
+        #     Delete received from AP                                         : 0
+        #     BSSID down                                                      : 1
+        #     AP down/disjoin                                                 : 2
+        #     Connection timeout                                              : 0
+        #     MAC authentication failure                                      : 0
+        #     Datapath plumb                                                  : 0
+        #     Due to SSID change                                              : 163
+        #
+        # ...OUTPUT OMITTED...
+        #
+        #     L3 VLAN Override connection timeout                         : 0
+        #     Mobility peer delete                                            : 0
+        #     NACK IFID mismatch                                              : 0
+
+        wlan_info_capture = re.compile(
+            # Wlan Profile Name: lizzard_Global, Wlan Id: 17
+            r"^Wlan Profile Name:\s+(?P<profile_name>\S+), Wlan Id: (?P<id>\d+)$"
+        )
+
+        # Current client state statistics:
+        client_stats_capture = re.compile(r"^Current client state statistics:$")
+
+        # Total client delete reasons
+        client_delete_capture = re.compile(r"^Total client delete reasons$")
+
+        # key : value
+        key_value_capture = re.compile(r"^(?P<key>[\S\s]+\S)\s*:\s+(?P<value>\d+)$")
+        
+        header_group = {}
+
+        wlan_info_obj = {}
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            if wlan_info_capture.match(line):
+                match = wlan_info_capture.match(line)
```

Comment:
I did a little testing for runtime and its about 4-5 times faster to do it this way:
``` python
m = wlan_info_capture.match(line)
if m:
    group = m.groupdict()
    (snip)
```
Time with two matches: 4.39999857917428e-05
Time with one match, saved to a variable: 8.999952115118504e-06

Reply:
4 secs is a lot! I appreciate the testing, I've updated with newest commit